### ISSUE TITLE
Playground POSTs to the wrong URI

### ISSUE THREAD

**Issue by curtisallen** [graphcool/graphql-playground#144]
While using express middleware after #139 

```javascript
const playground = require('graphql-playground/middleware').express
app.use('/playground', playground({ endpointUrl: '/graphql' }))
```

Playground will issue graphql queries to `/playground` instead of `/graphql`.

Playground is issuing the following request for the schema

<details>
<summary>curl 'http://localhost:8080/playground' </summary>
<p>
-H 'Pragma: no-cache' -H 'Origin: http://localhost:8080' -H 'Accept-Encoding: gzip, deflate, br' -H 'Accept-Language: en-US,en;q=0.8' -H 'Content-Type: application/json' -H 'Accept: */*' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36' -H 'Cache-Control: no-cache' -H 'Referer: http://localhost:8080/playground' -H 'Connection: keep-alive' -H 'x-graphcool-source: console:playground' --data-binary '{"query":"\n  query IntrospectionQuery {\n    __schema {\n      queryType { name }\n      mutationType { name }\n      subscriptionType { name }\n      types {\n        ...FullType\n      }\n      directives {\n        name\n        description\n        locations\n        args {\n          ...InputValue\n        }\n      }\n    }\n  }\n\n  fragment FullType on __Type {\n    kind\n    name\n    description\n    fields(includeDeprecated: true) {\n      name\n      description\n      args {\n        ...InputValue\n      }\n      type {\n        ...TypeRef\n      }\n      isDeprecated\n      deprecationReason\n    }\n    inputFields {\n      ...InputValue\n    }\n    interfaces {\n      ...TypeRef\n    }\n    enumValues(includeDeprecated: true) {\n      name\n      description\n      isDeprecated\n      deprecationReason\n    }\n    possibleTypes {\n      ...TypeRef\n    }\n  }\n\n  fragment InputValue on __InputValue {\n    name\n    description\n    type { ...TypeRef }\n    defaultValue\n  }\n\n  fragment TypeRef on __Type {\n    kind\n    name\n    ofType {\n      kind\n      name\n      ofType {\n        kind\n        name\n        ofType {\n          kind\n          name\n          ofType {\n            kind\n            name\n            ofType {\n              kind\n              name\n              ofType {\n                kind\n                name\n                ofType {\n                  kind\n                  name\n                }\n              }\n            }\n          }\n        }\n      }\n    }\n  }\n"}' --compressed
</p>
</details> 

Should be `curl 'http://localhost:8080/graphql'`

The UI reports the following error

```
SyntaxError: Unexpected token < in JSON at position 1
    at http://cdn.jsdelivr.net/npm/graphql-playground@1.0.3/middleware-build/static/js/main.js:23:28346
    at <anonymous>
```

<img width="785" alt="screen shot 2017-09-28 at 9 37 21 am" src="https://user-images.githubusercontent.com/270735/30976004-e39ee34a-a430-11e7-9549-8d4efa122b42.png">

**Comment by kkemple** [comment]
seeing this as well with hapi middleware using v1.0.3

config:
```javascript
{
  register: playground,
  options: {
    path: '/playground',
    endpointUrl: '/graphql',
  },
},
```

<img width="1676" alt="graphcool_playground" src="https://user-images.githubusercontent.com/3629876/30978930-09c0653a-a44a-11e7-8e6e-fd1725988a57.png">

**Comment by timsuchanek** [comment]
Sorry for the confusion, we updated the endpoint to `endpoint`. Can you check if it's fixed when renaming the option to `endpoint`?

**Comment by ecwyne** [comment]
@timsuchanek it looks like https://github.com/graphcool/graphql-playground/commit/ed5551696f65f7df4232116d4224d177da9f2de0 didn't update https://github.com/graphcool/graphql-playground/blob/master/packages/graphql-playground/src/middleware/render-playground-page.ts#L56

**Comment by curtisallen** [comment]
@timsuchanek yeah using `endpoint` didn't work
```javascript
app.use('/playground', playground({ endpoint: '/graphql' }))
```
Still posting to `/playground`

**Comment by overra** [comment]
It seems the problem is that the `endpoint` prop isn't getting passed down to the `<App>` component. I think we just need to use the `<Router>`'s [render](https://reacttraining.com/react-router/web/api/Route/render-func) prop and pass down the endpoint prop to fix this. Not sure if there's an easy way for me to test this since the build is hosted on a CDN. 

![image](https://user-images.githubusercontent.com/362590/31041031-b6a6d72e-a554-11e7-9c84-ccf2cb74dffa.png)

With the [React Developer Tools extension](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi) I was able to edit the endpoint prop and it works temporarily.

**Comment by overra** [comment]
After a few [fixes](https://github.com/graphcool/graphql-playground/pull/155) I was able to run this in development mode locally. Then I modified `public/index.html` so that it provided an endpoint option to `GraphQLPlayground.init`. I've made the changes I mentioned above and now it points to the custom endpoint.

**Comment by Glavin001** [comment]
This is disappointing. I was hoping to try out GraphQL Playground today and replace GraphiQL. Sounds like Playground is still unstable and I should wait a little longer. 

Playground looks great at https://www.graphqlbin.com! Keep up the awesome work. Hopefully a usable build will be released soon! 🎆 

For reference, I am also using:

```
const app = express();
app.use("/playground", playground({ endpoint: "/graphql" }));
```

with

```
    "graphql-playground": "1.0.4",
```

**Comment by timsuchanek** [comment]
It's now fixed in `1.0.8`

**Comment by curtisallen** [comment]
awesome works for me now thanks!
